str0m/src/media/mline.rs

```diff
@@ -367,23 +380,40 @@ impl MLine {
 
         // For resends, the original seq_no is inserted before the payload.
         let mut original_seq_len = 0;
-        if let Some(orig_seq_no) = next.orig_seq_no {
+        if let Some(orig_seq_no) = next.body.orig_seq_no() {
             original_seq_len = RtpHeader::write_original_sequence_number(body_out, orig_seq_no);
             body_out = &mut body_out[original_seq_len..];
         }
 
-        let body_len = next.pkt.data.len();
-        body_out[..body_len].copy_from_slice(&next.pkt.data);
+        match next.body {
+            NextPacketBody::Regular { pkt } | NextPacketBody::Resend { pkt, .. } => {
+                let body_len = pkt.data.len();
+                body_out[..body_len].copy_from_slice(&pkt.data);
+
+                // pad for SRTP
+                let pad_len = RtpHeader::pad_packet(
+                    &mut buf[..],
+                    header_len,
+                    body_len + original_seq_len,
+                    SRTP_BLOCK_SIZE,
+                );
 
-        // pad for SRTP
-        let pad_len = RtpHeader::pad_packet(
-            &mut buf[..],
-            header_len,
-            body_len + original_seq_len,
-            SRTP_BLOCK_SIZE,
-        );
+                buf.truncate(header_len + body_len + original_seq_len + pad_len);
+            }
+            NextPacketBody::Padding { len } => {
+                let len = RtpHeader::create_padding_packet(
+                    &mut buf[..],
+                    len,
+                    header_len,
+                    SRTP_BLOCK_SIZE,
+                );
+                if len == 0 {
+                    return None;
+                }
 
-        buf.truncate(header_len + body_len + original_seq_len + pad_len);
+                buf.truncate(header_len + len);
+            }
+        }
```

Comment:
Maybe move `buf.truncate`  to below and let the whole `match` return the body len

```rs
let body_len = match next.body {
 ...
}

buf.truncate(header_len + body_len);
```

Reply:
Sounds good 👍🏼